bspump/elasticsearch/connection.py

```diff
@@ -607,9 +600,31 @@ def build_headers(username, password, api_key):
 			'Content-Type': 'application/json',
 			"Authorization": 'ApiKey {}'.format(api_key)
 		}
-		# L.log(asab.LOG_NOTICE, 'Building headers with api_key')
 	else:
 		auth = None
 		Headers = None
 
 	return Headers, auth
+
+def get_credentials_from_config(configuration):
+	"""
+	Parses configurations to return credentials
+	"""
+
+	# new configuration
+	username = configuration.get('username')
+	password = configuration.get('password')
+	api_key = configuration.get('api_key')
+	url = configuration.get('url')
+
+	# backward compatibility with the format http://<username>:<password>@path
+	auth_credentials = configuration.get('elasticsearch_url')
+	if auth_credentials is not None:
+		asab.LogObsolete.warning("The configuration of ESConnection is outdated, please update the [connection:] section.")
```

Comment:
Add `struct_data: {"eof": "..."}` and provide more detailed description of what is wrong with the configuration.
The section is not named `[connection:]`.